core/templates/dev/head/pages/exploration_player/TutorCardDirective.js

```diff
@@ -190,56 +177,52 @@ oppia.directive('tutorCard', [
             $scope.onChangeInteractionAnswerValidity({
               answerValidity: answerValidity
             });
-            if (
-              ExplorationPlayerStateService.getCurrentInteraction().id ===
-              'CodeRepl') {
+            if ($scope.displayedCard.getInteraction().id === 'CodeRepl') {
               $scope.onChangeInteractionAnswerValidity({
                 answerValidity: true
               });
             }
           };
 
+          $scope.showAudioBar = function() {
+            return (
+              !$scope.isIframed &&
+              !ExplorationPlayerStateService.isInPretestMode());
+          };
+
           $scope.isContentAudioTranslationAvailable = function() {
             if ($scope.conceptCardIsBeingShown) {
               return false;
             }
             return (
-              ExplorationPlayerStateService.
-                isContentAudioTranslationAvailable());
+              $scope.displayedCard.isContentAudioTranslationAvailable());
           };
 
           $scope.isCurrentCardAtEndOfTranscript = function() {
-            return PlayerTranscriptService.isLastCard(
-              PlayerPositionService.getActiveCardIndex());
+            return !$scope.getDisplayedCard().isCompleted();
           };
 
           $scope.isOnTerminalCard = function() {
-            if ($scope.interactionIsActive) {
-              return (
-                ExplorationPlayerStateService.isCurrentStateTerminal());
-            }
             return (
-              INTERACTION_SPECS[$scope.activeCard.getInteraction().id].
-                is_terminal);
+              $scope.displayedCard.isTerminal());
```

Comment:
I'm seeing $scope.getDisplayedCard() and $scope.displayedCard in this file. Perhaps we should be consistent?

Reply:
Done.